### PR TITLE
add support for templated extraEnvironmentVariables

### PR DESCRIPTION
**Pull request by properpanda**
this is a small changeset that allows templating of extraEnvironmentVariables.

this will support using things like .Values values in extraEnvironmentVariables

### CODE REVIEW

**Comment by jainishshah17** [comment]
@properpanda Please update readme as well.

**Comment by properpanda** [comment]
> @properpanda Please update readme as well.

updated

File: stable/artifactory-ha/README.md [hunk 1]
```diff
@@ -556,7 +556,7 @@ The following table lists the configurable parameters of the artifactory chart a
 | `artifactory.internalPort`   | Artifactory service internal port                         | `8081`                        |
 | `artifactory.internalPortReplicator` | Replicator service internal port | `6061`   |
 | `artifactory.externalPortReplicator` | Replicator service external port | `6061`   |
-| `artifactory.extraEnvironmentVariables`          | Extra environment variables to pass to Artifactory. See [documentation](https://www.jfrog.com/confluence/display/RTF/Installing+with+Docker#InstallingwithDocker-SupportedEnvironmentVariables) |   |
+| `artifactory.extraEnvironmentVariables`          | Extra environment variables to pass to Artifactory. Supports supports evaluating strings as templates via the [`tpl`](https://helm.sh/docs/charts_tips_and_tricks/#using-the-tpl-function) function. See [documentation](https://www.jfrog.com/confluence/display/RTF/Installing+with+Docker#InstallingwithDocker-SupportedEnvironmentVariables) |   |
```
Comment (eldada) [inline stable/artifactory-ha/README.md]: Duplicate `supports`

File: stable/artifactory/README.md [hunk 2]
```diff
@@ -454,7 +454,7 @@ The following table lists the configurable parameters of the artifactory chart a
 | `artifactory.accessAdmin.dataKey`                | Artifactory access-admin secret data key |                                                                    |
 | `artifactory.preStartCommand`                    | Command to run before entrypoint starts |                             |
 | `artifactory.postStartCommand`                   | Command to run after container starts   |                             |
-| `artifactory.extraEnvironmentVariables`          | Extra environment variables to pass to Artifactory. See [documentation](https://www.jfrog.com/confluence/display/RTF/Installing+with+Docker#InstallingwithDocker-SupportedEnvironmentVariables) |   |
+| `artifactory.extraEnvironmentVariables`          | Extra environment variables to pass to Artifactory. Supports supports evaluating strings as templates via the [`tpl`](https://helm.sh/docs/charts_tips_and_tricks/#using-the-tpl-function) function. See [documentation](https://www.jfrog.com/confluence/display/RTF/Installing+with+Docker#InstallingwithDocker-SupportedEnvironmentVariables) |   |
```
Comment (eldada) [inline stable/artifactory/README.md]: Duplicate `supports`

**Pull request merged**